src/Directory.Build.targets

```diff
@@ -31,6 +31,11 @@
 		<XamlMergeInput Include="@(Page)" Exclude="@(_NonMergedXamlResources)" />
 	</ItemGroup>
 
+	<ItemGroup>
+      <Compile Remove="**/Generated/**/*.cs" />
+	  <Compile Include="**/Generated/**/*.cs" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
+	</ItemGroup>
```

Comment:
I don't see a problem, though I think it should placed in individual projects, not necessarily globally. Do you know why the order has an impact?

Reply:
> though I think it should placed in individual projects

It will need to be added in many places which I wanted to avoid.

Reply:
I would have though 3 projects, inside directory.build.props? I'd want to avoid altering projects that don't use uno sync generator.

Reply:
I was discussing this issue with Roslyn team, and the suggestion I got is to not mark these files as auto generated and let all analyzers run on them.

What do you think?
Instead of removing `// <auto-generated>` that was added recently to all generated files, I'll overwrite that with `.editorconfig`

Reply:
@jeromelaban The suggestion from Roslyn team to mark these files as non-generated doesn't look viable for us.

If we wanted to move this from `Directory.Build.targets`, we need to add it to the following projects:

- Uno.Foundation.Reference.csproj
- Uno.Foundation.Skia.csproj
- Uno.Foundation.Tests.csproj
- Uno.Foundation.Wasm.csproj
- Uno.Foundation.netcoremobile.csproj
- Uno.UI.Composition.Reference.csproj
- Uno.UI.Composition.Skia.csproj
- Uno.UI.Composition.Tests.csproj
- Uno.UI.Composition.Wasm.csproj
- Uno.UI.Composition.netcoremobile.csproj
- Uno.UI.Reference.csproj
- Uno.UI.Skia.csproj
- Uno.UI.Tests.csproj
- Uno.UI.Wasm.csproj
- Uno.UI.netcoremobile.csproj
- Uno.Reference.csproj
- Uno.Skia.csproj
- Uno.Tests.csproj
- Uno.Wasm.csproj
- Uno.netcoremobile.csproj

I wanted to avoid having to have this change in many places like that.

Reply:
Makes sense. We could use multiple explicit globbing in the top-level Directory.Build.targets, but it's likely going to increase the globbing performance impact, as it's matching the whole tree. It's likely going to already be a performance problem with the two statements as the PR currently is.

Wouldn't it be easier/more performance to adjust the generation so that it does not fail on currently active analyzers?